src/platform/EFR32/GroupKeyStoreImpl.cpp

```diff
@@ -1,386 +0,0 @@
-/*
- *
- *    Copyright (c) 2020 Project CHIP Authors
- *    Copyright (c) 2019 Nest Labs, Inc.
- *
- *    Licensed under the Apache License, Version 2.0 (the "License");
- *    you may not use this file except in compliance with the License.
- *    You may obtain a copy of the License at
- *
- *        http://www.apache.org/licenses/LICENSE-2.0
- *
- *    Unless required by applicable law or agreed to in writing, software
- *    distributed under the License is distributed on an "AS IS" BASIS,
- *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- *    See the License for the specific language governing permissions and
- *    limitations under the License.
- */
-
-/**
- *    @file
- *          Provides an implementation of the CHIP GroupKeyStore interface
- *          for platforms based on the Silicon Labs SDK.
- */
-/* this file behaves like a config.h, comes first */
-#include <platform/internal/CHIPDeviceLayerInternal.h>
-
-#include <platform/EFR32/GroupKeyStoreImpl.h>
-
-using namespace ::chip;
-using namespace ::chip::Profiles::Security::AppKeys;
-
-namespace chip {
-namespace DeviceLayer {
-namespace Internal {
-
-CHIP_ERROR GroupKeyStoreImpl::RetrieveGroupKey(uint32_t keyId, ChipGroupKey & key)
-{
-    CHIP_ERROR err;
-
-    // Iterate over all the GroupKey nvm3 records looking for a matching key...
-    err = ForEachRecord(kConfigKey_GroupKeyBase, kConfigKey_GroupKeyMax, false,
-                        [keyId, &key](const Key & nvm3Key, const size_t & length) -> CHIP_ERROR {
-                            CHIP_ERROR err2;
```

Comment:
Ironically, why I agree this is stale code, the group key management in Matter is based on Weave which is the base of this file, and this separation of business logic into a module would be good for the currently developed Group Key Management cluster. Still think these files cannot be used as-is, but it's too bad we did not get to this earlier, since the basis of group key management would have had a leg up.

Reply:
Yes, group key management is implemented as profile in Weave, but as cluster with a completely different APIs in CHIP.